corehq/apps/app_manager/models.py

```diff
@@ -4230,7 +4230,7 @@ def is_remote_app(self):
         return False
 
     @memoized
-    def get_previous_version(self):
+    def get_latest_build(self):
```

Comment:
:100:

Reply:
I think this rename is misleading. Since this called on a saved build in the middle of making a new build, arguably the latest build is `self` and the return value is the penultimate build (not that I suggest naming the function that). Is `get_previous_version` confusing?

Reply:
> Is `get_previous_version` confusing?

I think so - isn't `build` a clearer term?  Or would you say `version` is an acceptable synonym?  I'd be okay with `get_previous_build` though.

> arguably the latest build is `self`

Is that still true if the build isn't saved or finished yet?  This method looks up the build of `master_id` with the highest version in the DB at the time it's called.  I suppose memoization could make this confusing since at the end of the build process, there will be a new build with a higher version.

Reply:
`get_previous_build` sounds good, I agree it's clearer than `get_previous_version`.

> Is that still true if the build isn't saved or finished yet?

It's at least a little ambiguous, I think. Just realized I find `get_latest_build` confusing partly because I'm currently working on multi master, which (surprise!) adds a call to `get_previous_version` while pulling a linked app. So while I think right now `get_previous_version` is only called while making a build, that won't be true in the future.

Reply:
What's the multi master context that makes this confusing?  I'd think this would be even more clear outside the context of a build.

Another argument against "previous" is that it isn't correct when called on a build.  To me, "previous_build" means "the build before this one", but this method actually fetches the latest one regardless of whether it's called from the canonical app doc or an old build.

Reply:
In the multi master code, there's no ambiguity, the latest build is definitely `self`. This is called during the pull process, on a fully created/saved build, and is looking for the previous build.

Agreed that "previous" isn't accurate when this is called on a build (fwiw, I believe this is only ever called on builds).

Reply:
So `get_latest_build` is fine?
That name seems fine to me considering what it returns.. If its being used at places with a different intent, then those places need to update.